src/main/kotlin/org/jitsi/nlj/transform/node/PacketParser.kt

```diff
@@ -17,11 +17,23 @@ package org.jitsi.nlj.transform.node
 
 import org.jitsi.nlj.PacketInfo
 import org.jitsi.rtp.Packet
+import org.jitsi.utils.logging2.Logger
+
+open class PacketParser(
+    name: String,
+    parentLogger: Logger,
+    private val action: (Packet) -> Packet
+) : TransformerNode(name) {
+    private val logger = parentLogger.createChildLogger(javaClass.name)
 
-open class PacketParser(name: String, private val action: (Packet) -> Packet) : TransformerNode(name) {
     override fun transform(packetInfo: PacketInfo): PacketInfo? {
-        packetInfo.packet = action(packetInfo.packet)
-        packetInfo.resetPayloadVerification()
-        return packetInfo
+        return try {
+            packetInfo.packet = action(packetInfo.packet)
+            packetInfo.resetPayloadVerification()
+            packetInfo
+        } catch (e: Exception) {
+            logger.warn("Error parsing packet: $e")
+            null
```

Comment:
I think this should call `packetDiscarded` to count the packet as discarded and return the buffer to the pool

Reply:
Yes, good catch!